### PR TITLE
Add SystemStatus

### LINKED ISSUE #5: Define Astarte Interface for SystemStatus

### PR DESCRIPTION
**Pull request by FrancescoVaiani**
This commits add an interface with the porpoise of expose a `top` like data set , such as uptime and available memory.

Fix #5

Signed-off-by: Francesco Vaiani <francesco.vaiani@secomind.com>

### CODE REVIEW

File: io.edgehog.devicemanager.OSRuntimeStatus.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "interface_name": "io.edgehog.devicemanager.OSRuntimeStatus",
```
Comment (bettio) [inline io.edgehog.devicemanager.OSRuntimeStatus.json:2]: I was thinking about it, and maybe SystemStatus might be a better naming.

**Review by bettio — CHANGES_REQUESTED** [review]

File: io.edgehog.devicemanager.OSRuntimeStatus.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  "version_major": 0,
+  "version_minor": 1,
+  "type": "datastream",
+  "ownership": "device",
+  "aggregation": "object",
+  "mappings": [
+    {
```
Comment (bettio) [inline io.edgehog.devicemanager.OSRuntimeStatus.json:9]: let's add a TTL from day 0. I was thinking 60 days as default

**Review by bettio — CHANGES_REQUESTED** [review]

File: io.edgehog.devicemanager.OSRuntimeStatus.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      "endpoint": "/systemStatus/availMemory",
+      "type": "longinteger",
+      "explicit_timestamp": true,
+      "description": "Available memory (Bytes)",
+      "database_retention_policy": "use_ttl",
+      "database_retention_ttl": 5184000
+    },
+    {
+      "endpoint": "/systemStatus/bootId",
```
Comment (bettio) [inline io.edgehog.devicemanager.OSRuntimeStatus.json:18]: I think an uuid might be better.

**Review by bettio — CHANGES_REQUESTED** [review]

File: io.edgehog.devicemanager.OSRuntimeStatus.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{
+  "interface_name": "io.edgehog.devicemanager.SystemStatus",
+  "version_major": 0,
+  "version_minor": 1,
+  "type": "datastream",
+  "ownership": "device",
+  "aggregation": "object",
+  "mappings": [
+    {
+      "endpoint": "/systemStatus/availMemory",
```
Comment (bettio) [inline io.edgehog.devicemanager.OSRuntimeStatus.json:10]: availMemoryBytes

File: io.edgehog.devicemanager.OSRuntimeStatus.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    {
+      "endpoint": "/systemStatus/taskCount",
+      "type": "integer",
+      "explicit_timestamp": true,
+      "description": "Number of running tasks or processes",
+      "database_retention_policy": "use_ttl",
+      "database_retention_ttl": 5184000
+    },
+    {
+      "endpoint": "/systemStatus/uptime",
```
Comment (bettio) [inline io.edgehog.devicemanager.OSRuntimeStatus.json:34]: uptimeMillis

File: io.edgehog.devicemanager.OSRuntimeStatus.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      "type": "longinteger",
+      "explicit_timestamp": true,
+      "description": "Get time in minutes since boot",
```
Comment (bettio) [inline io.edgehog.devicemanager.OSRuntimeStatus.json:37]: I think we might go with milliseconds, since we can get them for free.

**Review by bettio — CHANGES_REQUESTED** [review]

**Review by bettio — CHANGES_REQUESTED** [review]
I would rephrase the commit message: `This commits add an interface with the porpoise of expose a top like data set , such as uptime and available memory.` to `Add interface for publishing `top`-like information, such as uptime and available memory.`

By the way, since we are not fixing an issue we might change Fix to Closes.

**Review by rbino — APPROVED** [review]
